src/entities/dhedge.ts

```diff
@@ -105,4 +105,13 @@ export class Dhedge {
       this.factory
     );
   }
+
+  /**
+   * Check if pool address is valid
+   * @param {string} address Pool address
+   * @returns {boolean} Is valid pool address
+   */
+  async validatePool(address: string): Promise<boolean> {
+    return await this.factory.isPool(address);
+  }
```

Comment:
in general, you don't need `async/await` here

Reply:
I know it's a contract call, but in this function body, it does nothing other than just passing `this.factory.isPool(address)` this Promise to its invoker. For simplicity, it is just unnecessary to use `async/await`.